runtime/hsa-runtime/core/driver/xdna/amd_xdna_driver.cpp

```diff
@@ -316,6 +342,22 @@ hsa_status_t XdnaDriver::InitDeviceHeap() {
   return HSA_STATUS_SUCCESS;
 }
 
+std::unordered_map<uint32_t, void*>& XdnaDriver::GetHandleMappings() {
+  return this->vmem_handle_mappings;
+}
+
+std::unordered_map<void*, uint32_t>& XdnaDriver::GetAddrMappings() {
+  return this->vmem_addr_mappings;
+}
+
+uint32_t XdnaDriver::GetTimeout() {
+  return this->timeout_val;
+}
+
+int XdnaDriver::GetFd() {
```

Comment:
If you move things that call into the kernel-mode driver into the XdnaDriver then we can avoid this. I hope we don't need to report the FD outside of the user-mode driver.

Reply:
Yep, that will allow this to be removed.

Reply:
Moved the work submission logic from the queue to the driver in 530dfee, which removed the need for this.